src/assets/scripts/client/aircraft/AircraftCollection.js

```diff
@@ -0,0 +1,213 @@
+     *
+     * @for AircraftCollection
+     * @method addItem
+     * @param aircraftModel
+     */
+    addItem(aircraftModel) {
+        // TODO: add instanceof check here
+        this._items.push(aircraftModel);
+        // FIXME: TOTAL HACK!!! REMOVE ME
+        prop.aircraft.list.push(aircraftModel);
+    }
+
+    /**
+     * @for AircraftCollection
+     * @method findAircraftDefinitionModelByIcao
+     * @param icao {string}
+     * @return {AircraftDefinitionModel}
+     */
+    findAircraftDefinitionModelByIcao(icao) {
+        return _find(this.definitionList, { icao: icao });
+    }
+
+    /**
+     * Loop through aircraft defined in the `definitionList` and create an
+     * `AircraftDefinitionModel` for each.
+     *
+     * @for AircraftCollection
+     * @method _buildAircraftDefinitionList
+     * @param aircraftDefinitionList {array}
+     * @return definitionList {array<AircraftDefinitionModel>}
+     * @private
+     */
+    _buildAircraftDefinitionList(aircraftDefinitionList) {
+        const definitionList = _map(aircraftDefinitionList, (aircraftDefinition) => {
+            // this is not using a direct return simply for readability
+            return new AircraftDefinitionModel(aircraftDefinition);
+        });
+
+        return definitionList;
+    }
+
+    /**
+     * Used to build up the appropriate data needed to instantiate an `AircraftInstanceModel`
+     *
+     * @for AircraftCollection
+     * @method _buildAircraftProps
+     * @param spawnModel {SpawnPatternModel}
+     * @return {object}
+     * @private
+     */
+    _buildAircraftProps(spawnModel) {
+        const airlineId = spawnModel.getRandomAirlineForSpawn();
+        const airlineModel = this._airlineCollection.findAirlineById(airlineId);
+        const aircraftDefinition = this._getAircraftDefinitionForAirlineId(airlineModel);
+        const destination = this._findDestinationFromRouteCode(spawnModel);
+        const { fleet } = airlineNameAndFleetHelper([airlineId]);
+        // TODO: move this to `AirlineModel`
+        const callsign = `${_random(0, 999)}`;
+
+        return {
+            callsign,
+            destination,
+            fleet,
+            category: spawnModel.category,
+            airline: airlineModel.icao,
+            altitude: spawnModel.altitude,
+            speed: spawnModel.speed,
+            heading: spawnModel.heading,
+            position: spawnModel.position,
+            icao: aircraftDefinition.icao,
+            model: aircraftDefinition,
+            route: spawnModel.route,
+            waypoints: _get(spawnModel, 'fixes', [])
+        };
+    }
+
+    /**
+     * Given an `airlineId`, find a random aircraft type from the airline.
+     *
+     * @for AircraftCollection
+     * @method _getAircraftDefinitionForAirlineId
+     * @param airlineId {string}
+     * @return aircraftDefinition {AircraftDefinitionModel}
+     * @private
+     */
+    _getAircraftDefinitionForAirlineId(airlineModel) {
```

Comment:
would you be ok with `AircraftTypeDefinition`?  Because that is essentially what this is, a type definition and I want to convey that with the name here.  Also, `type` has a very specific meaning in programming that is not applicable here.  so I think `definitionType` is where we'd need to go with this.